lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+						var tuple = objectCounts[type];
+						// Don't output events for no change? TODO - What's the
+						// correct
+						// behaviour here?
+						// TODO - Should we skip "weird" object types?
+						if (tuple['count'] != 0 && tuple['size'] != 0) {
+							callback("object." + type + ".count",
+									tuple['count']);
+							callback("object." + type + ".size", tuple['size']);
+						}
+					}
+					// fire events
+				}
+
+			}, objectTrackingCallbackInterval);
+}
+
+function start() {
+
+}
+
+function on() {
+	
+}
+
+function internal_on() {
+	
+}
+
+
+function profile(arg1, arg2, arg3) {
+
+}
+
+function configure(options) {
+
+}
+
+// CPU Profiling
+function startCpuProfiling() {
+	// If we aren't profiling already
+	if (!profiling) {
+		profiling = true;
+
+		// Instruct appmetrics to return JSON formatting and enable profiling
+		appmetrics.sendControlCommand("profiling_node",
+				"on,profiling_node_v8json");
+		appmetrics.enable('profiling');
+
+	}
+}
+
+function stopCpuProfiling(callback) {
+	if (profiling) {
+		
+		// Listen for JSON coming back from appmetrics
+		amapi.on('profiling', function(data) {
+			callback(JSON.stringify(data));
+		});
+
+		profiling = false;
+		appmetrics.disable('profiling');
+	}
+}
+
+// Object tracking
+function startTrackingObjects() {
+	objectTrackingEnabled = true;
+}
+
+function stopTrackingObjects() {
+	objectTrackingEnabled = false;
+	lastObjectHistogram = {};
+}
+
+// ---------------------------------------------
+// Private functions
+// ---------------------------------------------
+
+// Metrics API
+function createEventLoopListener() {
+	amapi.on('eventloop', function(res) {
+
+		var latency = res.latency;
+
+		// Update min and max accordingly
+		if (eventLoopMetrics.min == null) {
+			eventLoopMetrics.min = latency.min;
+		} else if (latency.min < eventLoopMetrics.min) {
+			eventLoopMetrics.min = latency.min;
+		}
+
+		if (eventLoopMetrics.max == null) {
+			eventLoopMetrics.max = latency.max;
+		} else if (latency.max > eventLoopMetrics.max) {
+			eventLoopMetrics.max = latency.max;
+		}
+
+		eventLoopMetrics.count++;
+		eventLoopMetrics.averageRunningTotal += latency.avg;
+	});
+}
+
+function createCpuListener() {
+	amapi.on('cpu', function(cpu) {
+		cpuMetrics.system += (cpu.system * 100);
+		cpuMetrics.user += (cpu.process * 100);
+		cpuMetrics.count++;
+	});
+}
+
+function createHeapListener() {
+	amapi.on('heap', function(res) {
+		heapMetrics.total = +res.total;
+		heapMetrics.used = +res.used;
+		heapMetrics.count++;
+	});
+}
+
+function createGCListener() {
+	amapi.on('gc', function(res) {
+		gcMetrics.used = +res.used;
+		gcMetrics.count++;
+	});
+}
+
+function createHTTPListener() {
+	amapi.on('http', function(res) {
+		// All we need to do is increment the count
+		httpMetrics.count++;
+	});
+}
+
+function createMessageListener() {
+	// Map the event types from axon and strong mq to whether they 
+	// were send or receive events.
+	var incrementCounts = function (data) {
```

Comment:
please just define it as a function: `function incrementCounts(data) {`, and define it after use at the end of the function - its more understandable to read code in the order it is executed: `.on(blah, fu); function fu() { ...} `